libbpf-tools/slabratetop.bpf.c

```diff
@@ -0,0 +1,54 @@
+/* SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause) */
+/* Copyright (c) 2022 Rong Tao */
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+#include "slabratetop.h"
+#include "stat.h"
+
+#define MAX_ENTRIES	10240
+
+const volatile pid_t target_pid = 0;
+
+static struct slabrate_info slab_zero_value = {};
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_ENTRIES);
+	__type(key, char *);
+	__type(value, struct slabrate_info);
+} slab_entries SEC(".maps");
+
+static int probe_entry(struct kmem_cache *cachep)
+{
+	__u64 pid_tgid = bpf_get_current_pid_tgid();
+	__u32 pid = pid_tgid >> 32;
+	struct slabrate_info *valuep;
+    const char *name = BPF_CORE_READ(cachep, name);
+
+	if (target_pid && target_pid != pid)
+		return 0;
+
+	valuep = bpf_map_lookup_elem(&slab_entries, &name);
+	if (!valuep) {
+		bpf_map_update_elem(&slab_entries, &name, &slab_zero_value, BPF_ANY);
+		valuep = bpf_map_lookup_elem(&slab_entries, &name);
```

Comment:
> Use `bpf_map_lookup_or_try_init` defined in `<maps.bpf.h>`

I think `bpf_map_lookup_or_try_init` is not suit for this situation, because i need initialize some field like `valuep->name` in `if (!valuep) {`, but can't in `slab_zero_value` do that, otherwise i need execute `bpf_probe_read_kernel(&slab_zero_value.name, sizeof(slab_zero_value.name), name);` every time `probe_entry` called.  This situation same as [filetop](https://github.com/iovisor/bcc/blob/master/libbpf-tools/filetop.bpf.c).